subiquity/common/apidef.py

```diff
@@ -412,8 +414,16 @@ def GET() -> Optional[ADConnectionInfo]: ...
         # simultaneously - such as invalid chars on the admin name and DC name
         # starting with a hyphen or a dot. Thus this must returns a List
         # of errors [ADValidationResult.OK] on success.
-        def POST(data: Payload[ADConnectionInfo]) \
-            -> List[ADValidationResult]: ...
+        def POST(data: Payload[ADConnectionInfo]) -> None: ...
+
+        class check_domain_name:
+            def GET(domain_name: str) -> List[AdDomainNameValidation]: ...
+
+        class check_admin_name:
+            def GET(admin_name: str) -> List[AdAdminNameValidation]: ...
+
+        class check_password:
+            def GET(password: str) -> AdPasswordValidation: ...
```

Comment:
AFAIK this prototype means the password is passed in the query string, so might end up in the access logs. You might want to use Payload[str] so it is passed in the body instead.

It is less of a problem for the other calls, but it feels a bit redundant when you read it:

GET /check_admin_name?admin_name=...
GET /check_domain_name?domain_name=...